plugins/hypervisors/hyperv/src/com/cloud/hypervisor/hyperv/resource/HypervDirectConnectResource.java

```diff
@@ -595,6 +602,25 @@ private PlugNicAnswer execute(final PlugNicCommand cmd) {
         }
     }
 
+    private CopyFileInVmAnswer execute(CopyFileInVmCommand cmd) {
+        File keyFile = getSystemVMKeyFile();
+        try {
+            File file = new File(cmd.getSrc());
+            if(file.exists()) {
```

Comment:
What about instantiating the response right away `CopyFileInVmAnswer copyFileInVmAnswer = new CopyFileInVmAnswer(cmd)` in the first lines? Then, you could reduce this cyclomatic complexity here (if/if/for). You could negate the Ifs, if the negation is true, you then return `copyFileInVmAnswer `.

what do you think of that?